### ISSUE TITLE
[package] xorg/system: false positive of missing requirements: xtrans-dev xkb-data

### ISSUE THREAD

**Issue by GavinNL** [conan-io/conan-center-index#13588]
### Description

While trying to run conan install on Ubuntu 22.04 docker/podman image, the `xorg/system` package started giving false positive errors for missing system packages: xtrans-dev and xkb-data.

This only started happening in the last few days as our CI was failing for Ubuntu 22.04 builds. Ubuntu 18.04 builds don't seem to be affected.

Below is the error I was getting along with proof that the "missing" packages are installed

I am running this in a isolated podman container based off ubuntu 22.04. 

My Linux Mint 21 (based off Ubuntu 22.04) seems to be working though.

```bash
xorg/system: ERROR: while executing system_requirements(): System requirements: 'xtrans-dev, xkb-data' are missing but can't install because tools.system.package_manager:mode is 'check'.Please update packages manually or set 'tools.system.package_manager:mode' to 'install' in the [conf] section of the profile, or in the command line using '-c tools.system.package_manager:mode=install'
ERROR: Error in system requirements

user@939acd8a9ea7:~/build$ sudo apt install xtrans-dev xkb-data
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
xkb-data is already the newest version (2.33-1).
xtrans-dev is already the newest version (1.4.0-1).
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.

```

### Package and Environment Details

* Package Name/Version: **xorg/system**
* Operating System+version: **Linux Ubuntu 22.04**
* Compiler+version: **GCC 11**
* Docker image:  Custom podman image based off ubuntu 22.04
* Conan version: **conan 1.53.0**
* Python version: **Python 3.10.6**

### Conan profile

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=11
compiler.libcxx=libstdc++
build_type=Release
[options]
[build_requires]
[env]

### Steps to reproduce

Create folder with the following conanfile.txt

```
[requires]
xorg/system

[options]

[generators]
cmake
cmake_find_package

[imports]
```

conan install PATH_TO_CONANFILE.TXT -s compiler.libcxx=libstdc++11 --build missing

### Logs

<details><summary>Click to expand log</summary>

```
user@939acd8a9ea7:~/build$ sudo apt install xtrans-dev xkb-data
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
xkb-data is already the newest version (2.33-1).
xtrans-dev is already the newest version (1.4.0-1).
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
user@939acd8a9ea7:~/build$ 
user@939acd8a9ea7:~/build$ 
user@939acd8a9ea7:~/build$ conan install ../project -s compiler.libcxx=libstdc++11 --build missing
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

conanfile.txt: Installing package
Requirements
    xorg/system from 'conancenter' - Cache
Packages
    xorg/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache

Installing (downloading, building) binaries...
dpkg-query: no packages found matching xtrans-dev:amd64
dpkg-query: no packages found matching xkb-data:amd64
xorg/system: ERROR: while executing system_requirements(): System requirements: 'xtrans-dev, xkb-data' are missing but can't install because tools.system.package_manager:mode is 'check'.Please update packages manually or set 'tools.system.package_manager:mode' to 'install' in the [conf] section of the profile, or in the command line using '-c tools.system.package_manager:mode=install'
ERROR: Error in system requirements
user@939acd8a9ea7:~/build$ 
```

</details>

**Comment by ericLemanissier** [comment]
This is linked to https://github.com/conan-io/conan/issues/12320

**Comment by lasersson** [comment]
So until that feature is implemented and we have all upgraded xorg and all it's dependents will remain broken? At the very least on the reported OS?

**Comment by jothalha** [comment]
As a hotfix we overwrote the xorg dependency with the `ac8a8a26d70a5a699802955008935941` revision before the introduction of `arch`. that restored our pipeline again.

**Comment by lasersson** [comment]
Ya, that would do it. I'm right now trying to figure out how to do just that when xorg is a transitive dependency, i.e. I'm not dependning on xorg directly but via vulkan-loader

**Comment by lasersson** [comment]
Having our builds tactically nuked by breaking changes to conan center recipes is becoming really concerning. This is not the first time this has happened.

**Comment by ericLemanissier** [comment]
Yes, breaking changes happen more often than what we all would like. A way to mitigate that is to follow the recommendations on https://github.com/conan-io/conan-center-index/blob/master/docs/consuming_recipes.md

**Comment by ericLemanissier** [comment]
I think the best course of action for this bug is to remove `xtrans` and `xkb-data` from `xog/system`, and create a `xtrans/system` and `xkb/system` package which do not have an `arch` setting. I'm working on it

**Comment by ericLemanissier** [comment]
I think the best course of action for this bug is to remove `xtrans` and `xkb-data` from `xog/system`, and create a `xtrans/system` and `xkb/system` package which do not have an `arch` setting. I'm working on it

**Comment by ericLemanissier** [comment]
You are not, this is a bug in my xorg/system recipe. I just removed the requirement on xtrans and xkeyboard-config. Can you please retry ?
It means that this requirement will need to be added to recipes actually using these libraries. I know xkbcommon uses xkeyboar-config. I'll do it.

**Comment by GavinNL** [comment]
Looks like it worked

```bash
root@713fb5dd1732:/home/user/conan/build# conan install .. -s compiler.libcxx=libstdc++11 --build missing
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

conanfile.txt: Installing package
Requirements
    xorg/system from local cache - Cache
Packages
    xorg/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache

Installing (downloading, building) binaries...
xorg/system: Already installed!
conanfile.txt: Generator txt created conanbuildinfo.txt
conanfile.txt: Generator cmake created conanbuildinfo.cmake
conanfile.txt: Generator cmake_find_package created Findxorg.cmake
conanfile.txt: Aggregating env generators
conanfile.txt: Generated conaninfo.txt
conanfile.txt: Generated graphinfo

```

**Comment by ericLemanissier** [comment]
thanks. does it also work with whatever downstream package you want to build in the end ?

**Comment by GavinNL** [comment]
Yes, seems to be working. This is the conanfile i'm using for my project in case you need it:

```
[requires]
stb/20200203
spdlog/1.5.0
glm/0.9.9.7
nlohmann_json/3.7.0
entt/3.6.0
catch2/2.13.6
tng/1.8.2
span-lite/0.9.0
pybind11/2.9.1
glbinding/3.1.0
lyra/1.5.1
#easy_profiler/2.1.0
sdl/2.0.20
nanoflann/1.3.2
cppzmq/4.6.0
rpclib/2.3.0
zlib/1.2.12
range-v3/0.11.0
exprtk/0.0.1
libxml2/2.9.14
libiconv/1.17
libalsa/1.2.7

[options]
sdl:jack=False
sdl:shared=True
sdl:wayland=False

[generators]
cmake
cmake_find_package
```

**Comment by ericLemanissier** [comment]
This can be closed, it was fixed with https://github.com/conan-io/conan-center-index/pull/13592

**Comment by uilianries** [comment]
Fixed by #13592
